### DISCUSSION TITLE
What about `sessionStorage`?

### DISCUSSION THREAD

**Discussion by jsmunroe** [sushinpv/react-secure-storage#38] [Ideas]
I had to fork this repo. [Here ](https://github.com/jsmunroe/react-secure-storage) is my fork. My team is using `sessionStorage` instead of `localStorage` because we want the data to live only for the duration of the user's session. 

Both of these types are `Storage` types and so I generalized your `SessionLocalStorage` class into a `SessionStorage` class that takes a `Storage` object in its constructor. This way the `sessionLocalStorage` instance is instantiated like this...

```JavaScript
const secureLocalStorage = new SecureStorage(localStorage);
```
I left `secureLocalStorage` as the default export to avoid breaking changes and added `secureSessionStorage` as an auxiliary export.  This instance is instantiated as such...

```JavaScript
const secureSessionStorage = new SecureStorage(sessionStorage);
```

`sessionStorage` is supported in [95.91%](https://caniuse.com/?search=sessionstorage) of browsers and adds a lot of flexibility when it comes to life spans of client-side data. Feel free to incorporate this change into your own repository, just let me know when you do so I can move my project back over to your package.

Thanks for the great code. Much luck.

**Comment by sushinpv** [comment]
@jsmunroe Thanks a lot for your contribution, Please give me sometime to go through your code and run a test against our usecases!

In the meantime, You can create a PR to the development, 

Thank you for your contribution

**Reply by jsmunroe** [reply]
Thank you for your quick response and your willingness to look into this commit. 

I think I did pull request the changes. I'm not sure. This is my first time collaborating on github. Let me know if I need to do something other than what I think I did. :)

**Reply by sushinpv** [reply]
Hi @jsmunroe We are testing your PR, once we have any updates we will let you know! 

Thanks for waiting

**Reply by Anon0x19** [reply]
Hi there @sushinpv, I noticed the [PR](https://github.com/sushinpv/react-secure-storage/pull/40) you raised hasn't been merged yet, and I'm not sure if it's been discontinued. I thought it might be good to mention it here.

I've raised a [PR](https://github.com/sushinpv/react-secure-storage/pull/49) to implement session storage. I used the PR you both raised as guidance when implementing it and successfully tested it with one of my projects.

I hope we can get session storage support merged soon, as it would be a very useful feature.

Best regards,
Marc
